src/bin/cargo.rs

```diff
@@ -91,7 +91,14 @@ macro_rules! each_subcommand{
         $mac!(verify_project);
         $mac!(version);
         $mac!(yank);
-    })
+    }
+}
+
+mod subcommands {
+    macro_rules! declare_mod {
+        ($name:ident) => ( pub mod $name; )
+    }
+    each_subcommand!(declare_mod);
```

Comment:
Yeah, it should require that. Right now, macro-expanded non-inline module declarations are always treated as if they were at the file root due to a bug in the macro expander.